### PR TITLE
FOGL-1869: Poll south plugins accurately

### PR DESCRIPTION
**Pull request by AmandeepArora**
Used timerfd_* routines (alternative to itimers) to poll south plugins accurately. Tried with random plugin and readings were generated very accurately. Even if timer expiry events are missed (e.g. say south service doesn't get scheduled for multiple pollIntervals), then this event backlog is handled to catch up with required rate.

Tried with 20, 50, 100 and 1000 msecs pollInterval and expected number of readings got generated very accurately despite multiple consecutive timer expiry events being missed.

### CODE REVIEW

**Comment by AmandeepArora** [comment]
Sep 26 13:53:24 foglamp-VirtualBox Random1[16859]: Created timer FD for 100000 usecs
Sep 26 13:54:03 foglamp-VirtualBox Random1[16859]: 390 readings generated in 39.1206546 secs
Sep 26 13:56:04 foglamp-VirtualBox Random1[17441]: Created timer FD for 100000 usecs
Sep 26 13:56:32 foglamp-VirtualBox Random1[17441]: 285 readings generated in 28.500211026 secs
Sep 26 13:56:48 foglamp-VirtualBox Random1[17504]: Created timer FD for 100000 usecs
Sep 26 13:57:40 foglamp-VirtualBox Random1[17504]: 523 readings generated in 52.316243437 secs
Sep 26 13:58:04 foglamp-VirtualBox Random1[17636]: Created timer FD for 10000 usecs
Sep 26 13:59:02 foglamp-VirtualBox Random1[17636]: 5836 readings generated in 58.360184439 secs
Sep 26 13:59:33 foglamp-VirtualBox Random1[17751]: Created timer FD for 50000 usecs
Sep 26 14:00:17 foglamp-VirtualBox Random1[17751]: 882 readings generated in 44.127066500 secs
Sep 26 14:00:53 foglamp-VirtualBox Random1[17859]: Created timer FD for 20000 usecs
Sep 26 14:22:11 foglamp-VirtualBox Random1[17859]: 63943 readings generated in 1279.70647356 secs
Sep 26 14:35:20 foglamp-VirtualBox Random1[20770]: Created timer FD for 20000 usecs
Sep 26 14:35:33 foglamp-VirtualBox Random1[20770]: 690 readings generated in 13.800995771 secs
Sep 26 14:37:30 foglamp-VirtualBox Random1[21109]: Created timer FD for 20000 usecs
Sep 26 14:40:30 foglamp-VirtualBox Random1[21109]: 8970 readings generated in 179.407041296 secs

File: C/services/south/south.cpp [hunk 1]
```diff
@@ -22,6 +26,7 @@
 #include <filter_plugin.h>
 
 extern int makeDaemon(void);
+static int createTimerFd(unsigned int usecs);
```
Comment (MarkRiddoch) [inline C/services/south/south.cpp]: Why a static function and not a private method

Comment (AmandeepArora) [reply C/services/south/south.cpp]: It's a helper function, which I feel could be used by other code also in future. It's not specific to SouthService class, hence didn't want to make it part of this class.

Comment (AmandeepArora) [reply C/services/south/south.cpp]: @MarkRiddoch - Please let me know if I should make this change or leave it as it is.

Comment (MarkRiddoch) [reply C/services/south/south.cpp]: It's up to you, personally I would make it a member function, but I don't think it is that much of an issue that I would stop the checkin

**Comment by AmandeepArora** [comment]
@MarkRiddoch - Made createTimerFd private function of SouthService class.

**Pull request merged**